r/R/dplyr.R

```diff
@@ -151,8 +151,14 @@ dim.arrow_dplyr_query <- function(x) {
     rows <- NA_integer_
   } else if (isTRUE(x$filtered_rows)) {
     rows <- x$.data$num_rows
-  } else {
+  } else if (query_on_dataset(x)) {
+    # TODO: do this with an ExecPlan instead of Scanner?
```

Comment:
```suggestion
    # TODO: do this with an ExecPlan instead of Scanner (after ARROW-12311)?
    # See also https://github.com/apache/arrow/pull/12533/files#r818129459
```